### PR TITLE
Comments - model and tests

### PR DESCRIPTION
**Pull request by gabrielkay**
### Issue
#10 

### Why?
This is the first step in setting up a threaded comment system

### What?
- Added comment model and db migration file
- Modified User and Post models to reflect their relationships with comments
- Added comment specific tests in comment_spec.rb

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    it { should validate_presence_of(:body) }
+  end
+
+  context 'comment with no user_id' do
+    let(:comment) { build(:comment, user_id: nil) }
+
+    it { should validate_presence_of(:user_id) }
+  end
+
+  context 'comment with no post_id' do
```
Comment (benjtinsley) [inline spec/models/comment_spec.rb]: maybe someone who understand rails better than me can fill me in here, but what are these tests verifying? how would there ever be a comment with no post id? it seems like what this is testing is whether the application can build a post with no id/comment/user. perhaps this is a method of testing that i am unfamiliar with, however.

the way i'd approach these is creating a user, and trying to submit a comment with no body and verifying that the page gives the adequate warning to let the user know that it is a disallowed action to do so.

Comment (gabrielkay) [reply spec/models/comment_spec.rb]: These are testing the comment model validator, to make sure it doesn't allow comments missing any of the required fields. This test using shoulda-matchers essentially does what you described in your second paragraph automatically (it creates generic users/posts/comments from the ones outlined in the factories folder is my understanding). I initially wrote out some tests that way but I changed them to this simpler format.

Comment (benjtinsley) [reply spec/models/comment_spec.rb]: do you need to verify the comment model? again, speaking from lack of experience here but it doesn't seem like a lot of custom domain logic you are adding to the model with these lines:
```ruby
  validates :post_id, presence: true
  validates :user_id, presence: true
  validates :body, presence: true``` 

the second paragraph implies there's some sort of notice helper configured to show the user warnings if they do a disallowed action and it also implies theres a route set up to post a comment to, in the event that it succeeds. perhaps it is not valuable to test it this way in this scenario, so forgive me if i'm focusing too much on this particular test.

Comment (benjtinsley) [reply spec/models/comment_spec.rb]: sorry, and i realize now that this is specifically in `spec/models` so testing model domain logic is given. i don't ever think i've written a model test, so nice to see how it's done

Comment (gabrielkay) [reply spec/models/comment_spec.rb]: A test I originally wrote for the post model:
```
it "can't create post without link" do
    post = build(:post, link: nil)
    post.valid?
    expect(post.errors[:base]).to include("You must submit a valid URL. You entered: ")
  end
```
I believe that `it { should validate_presence_of(:link) }` is logically equivalent and shoulda-matchers does all those steps internally which is pretty cool. @albaer could confirm that.

I agree there isn't much logic so these tests aren't doing _that_ much.

Comment (nhunzaker) [reply spec/models/comment_spec.rb]: 👍. I think of these as being less about testing functionality and more about testing that configuration is correct. From what I understand, the validations catch things application side and format database errors. If you don't have them configured correctly, you can get errors back from Postgres without a great way to tell a user what's going on.

`it { should validate_presence_of(:post_id) }` just makes that super easy.

**Pull request merged**